packages/runtime-core/src/componentRenderUtils.ts

```diff
@@ -327,14 +327,15 @@ export function shouldUpdateComponent(
 ): boolean {
   const { props: prevProps, children: prevChildren, component } = prevVNode
   const { props: nextProps, children: nextChildren, patchFlag } = nextVNode
-  const emits = component!.emitsOptions
 
   // Parent component's render function was hot-updated. Since this may have
   // caused the child component's slots content to have changed, we need to
   // force the child to update as well.
   if (__DEV__ && (prevChildren || nextChildren) && isHmrUpdating) {
     return true
   }
+  
+  const emits = component!.emitsOptions
```

Comment:
Why not just move it below the next if block, since we only use the value after that.

Reply:
I agree, I'd probably move it later too.

But just to make the case for the current version, I think it'll minify better in production builds as it'll keep the `const` declarations consecutive after dead-code removal. Moving it after the next `if` would prevent that.

Reply:
Updated